engine/engine/src/test/main.script

```diff
@@ -59,7 +59,8 @@ local function deactivate_proxy(self, index)
     msg.post(proxy, "unload")
 end
 
-function on_message(self, message_id, message)
+ --TODO: Rearrange this mess: Make sure each state is only started after the previous is unloaded!
+function on_message(self, message_id, message, sender)
```

Comment:
The loading logic seems flawed, sending disabled messaged to non-enabled collections.